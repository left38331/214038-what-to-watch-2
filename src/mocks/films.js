export const films = [
  {
    id: 1,
    title: `Fantastic Beats`,
    poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    alt: `Fantastic Beasts: The Crimes of Grindelwald`,
    link: `movie-page.html`,
    preview: `https://www.kinomania.ru/load/n?file=//fs.kinomania.ru/trailers/fbwft-rutlr3_854.mp4`,
    genre: `Fantasy`
  },
  {
    id: 2,
    title: `Bohemian Rhapsody`,
    poster: `img/bohemian-rhapsody.jpg`,
    alt: `Bohemian Rhapsody`,
    link: `movie-page.html`,
    preview: `https://www.kinomania.ru/load/n?file=//fs.kinomania.ru/media/video/c/76/c766c75666111cab4aa6bb75a5424bc0.480.mp4`,
    genre: `Drama`
  },
  {
    id: 3,
    title: `Macbeth.jpg`,
    poster: `img/macbeth.jpg`,
    alt: `Macbeth`,
    link: `movie-page.html`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Drama`
  },
  {
    id: 4,
    title: `Aviator`,
    poster: `img/aviator.jpg`,
    alt: `Aviator`,
    link: `movie-page.html`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genre: `Adventure`
  },
  {
    id: 5,
    title: `War of the worlds`,
    poster: `img/war-of-the-worlds.jpg`,
    alt: `War of the worlds`,
    link: `movie-page.html`,
    preview: `https://www.kinomania.ru/load/n?file=//fs.kinomania.ru/media/video/5/2a/52ab2988c90e81715008ed5189632b8d.480.mp4`,
    genre: `Action`
  },
];
