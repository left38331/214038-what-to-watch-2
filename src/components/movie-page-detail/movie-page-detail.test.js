import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import {Provider} from 'react-redux';

import MoviePageDetail from 'components/movie-page-detail/movie-page-detail';
import {films} from '../../mocks/films';

it(`Render correctly movie-page-detail component`, () => {
  const middleware = [thunk];
  const mockStore = configureMockStore(middleware);
  const initialState = {
    genre: `All genres`,
    listCardFilms: [],
    promo: {},
    activeFilms: [],
    genresList: [],
    isAuthorizationRequired: true,
    avatarUrl: ``,
    email: ``,
    id: ``,
    name: ``,
    film: ``,
    time: ``,
    comments: [],
    isPostComment: false,
    favoriteFilms: []
  };
  const store = mockStore(initialState);
  const props = {
    listCardFilms: films,
    history: {},
    match: {},
    clickTabHandler: ()=>{},
    toggleFavoriteFilm: ()=>{},
    setPlayingFilm: ()=>{},
    comments: [],
    label: `true`,
    isAuthorizationRequired: true,
    avatar: `avatar.jpg`,
    activeFilms: []
  };
  const moviePageDetail = renderer.create(<BrowserRouter><Provider store={store}><MoviePageDetail {...props}/></Provider></BrowserRouter>).toJSON();

  expect(moviePageDetail).toMatchSnapshot();
});
