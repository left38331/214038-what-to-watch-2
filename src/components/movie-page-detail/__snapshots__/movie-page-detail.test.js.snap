// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly movie-page-detail component 1`] = `null`;
